test/e2e/initscalezero/init_scale_zero_test.go

```diff
@@ -0,0 +1,140 @@
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := e2e.Setup(t)
+	tests := []struct {
+		name string
+		// configAutoscalerFlag indicates if the scale to zero on deploy flag is set to true
+		// or false in the config-autoscaler config map.
+		configAutoscalerFlag bool
+	}{{
+		name:                 "config-autoscaler true",
+		configAutoscalerFlag: true,
+	}, {
+		name:                 "config-autoscaler false",
+		configAutoscalerFlag: false,
+	}}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Logf("Setting flag to %v on config-autoscaler ConfigMap.", tc.configAutoscalerFlag)
+			if tc.configAutoscalerFlag {
+				updatedConfigAutoscalerCM := setScaleToZeroOnDeployOnCluster(t, clients, "config-autoscaler")
+				defer restoreCM(t, clients, updatedConfigAutoscalerCM)
+			}
+			names := test.ResourceNames{
+				Service: test.ObjectNameForTest(t),
+				Image:   "helloworld",
+			}
+			defer test.TearDown(clients, names)
+			t.Logf("Creating a new Service and verifying that scale to zero is %v.", tc.configAutoscalerFlag)
+			createServiceAndCheckPods(t, clients, names, tc.configAutoscalerFlag /* scaledToZero */)
+		})
+	}
+}
+
+// TestInitScaleZeroMinScaleClusterLevel tests setting of scaleToZeroOnDeploy on
+// config-autoscaler with minScale > 0
+func TestInitScaleZeroMinScaleClusterLevel(t *testing.T) {
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := e2e.Setup(t)
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "helloworld",
+	}
+	defer test.TearDown(clients, names)
+
+	updatedConfigAutoscalerCM := setScaleToZeroOnDeployOnCluster(t, clients, "config-autoscaler")
+	defer restoreCM(t, clients, updatedConfigAutoscalerCM)
+
+	t.Log("Creating a new Service with minScale greater than 0 and verifying that pods are created.")
+	createServiceAndCheckPods(t, clients, names, false,
+		v1a1testing.WithConfigAnnotations(map[string]string{
+			autoscaling.MinScaleAnnotationKey: "1",
+		}))
+}
+
+func restoreCM(t *testing.T, clients *test.Clients, oldCM *v1.ConfigMap) {
+	if _, err := e2e.PatchCM(clients, oldCM); err != nil {
+		t.Fatalf("Error restoring ConfigMap %q: %v", oldCM.Name, err)
+	}
+	t.Logf("Successfully restored ConfigMap %s", oldCM.Name)
+}
+
+func setScaleToZeroOnDeployOnCluster(t *testing.T, clients *test.Clients, configMapName string) *v1.ConfigMap {
+	t.Logf("Fetch %s ConfigMap", configMapName)
+	original, err := e2e.RawCM(clients, configMapName)
+	if err != nil || original == nil {
+		t.Fatalf("Error fetching %s: %v", configMapName, err)
+	}
+	var newCM *v1.ConfigMap
+	original.Data["scale-to-zero-on-deploy"] = "true"
+	if newCM, err = e2e.PatchCM(clients, original); err != nil {
+		t.Fatalf("Failed to update %s: %v", configMapName, err)
+	}
+	t.Logf("Successfully updated %s configMap.", configMapName)
+
+	newCM.Data["scale-to-zero-on-deploy"] = "false"
+	test.CleanupOnInterrupt(func() { restoreCM(t, clients, newCM) })
```

Comment:
Wonder if `test.CleanupOnInterrupt` should be at the top level `Test*` methods so that it's obvious what's being restored